usagov/usagov-2021
# Usagov 1685 fix site banner

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1685

## Description

 ~~I added a variable to Drupal's state to check if the banner is already placed or not. If that variable is `FALSE` and the banner is supposed to be visible, it runs the "drupal_flush_all_caches();" function.~~

I changed all the code for how we place a site banner but the interface and the process in which a content admin adds it remains the same. These were the things I did:
- Update the readme
- Remove the old code from `usa_twig_vars.module`
- Create a new module for the site_banner called `usa_site_banner`
- Remove the twig template that I was using to place the block 

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


1. Run `bin/init` and `bin/drush cim`
2. Create a Content Administrator account and log in using that account
3. Go to the Content blocks page `localhost/admin/content/block`
4. Click `+ Add content block`.
5. Fill in the required fields for the Site Banner.
   - Make sure you select the correct language for the banner.
6. Find the checkbox labeled `Place above header`.
   - Check this box if you want the banner to appear in the `header_top` region.
   - Uncheck this box if you want to remove the banner from the `header_top` region.
7. Click `Save` to create or update the Site Banner block.
8. Visit the site to ensure the banner appears or is removed from the region.
9. Log out from the account and check that the banner is visible for anonymous users. 
10. Unplaced the block and check it was removed from anonymous users as well.
11. Repeat these steps for both languages and also add multiple site_banners
12. Keep the banners placed and go to Docker Desktop and find the cms container
13. Click the 3 dots in the Actions column and click "Open in terminal"
14. In the terminal, type ./scripts/tome-run.sh and wait for tome to run
15. Once tome is done, go to http://127.0.0.1 and check that the banners that you placed are visible in all the pages.

### Change Requirements


- [x] Requires New Documentation (Link: {docs/SiteBanner.md})

### Validation Steps

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
